scripts/core/editor3/reducers/table.tsx

```diff
@@ -56,19 +73,28 @@ const addTable = (state, data) => {
  * @description Adds a row after the currently active one.
  */
 const addRowAfter = (state) =>
-    processCells(state, (prevCells, numCols, nRows, i, j, withHeader) => {
-        const numRows = nRows + 1;
-        const cells = [];
+    processCells(state, (tableData, activeCell) => {
+        const {i} = activeCell;
+        const {cells, numRows} = tableData;
+
 
-        prevCells.forEach((row, index) => {
-            cells.push(row);
+        const nextCells = [];
+
+        cells.forEach((row, index) => {
+            nextCells.push(row);
 
             if (index === i) {
-                cells.push([]);
+                nextCells.push([]);
             }
         });
 
-        return {data: {cells, numRows, numCols, withHeader}};
+        return {
+            data: {
+                ...tableData,
+                cells: nextCells,
+                numRows: numRows + 1,
```

Comment:
Don't leave magic numbers, I know it's just adding an additional role, but would be easier to read if it's something like + `ONE_ADDITIONAL_ROW`

Reply:
It's not magic here, because the purpose is clear and it couldn't work with any other number. It makes sense to use constants when it's arbitrary numbers e.g. items per page in paginated content.